util/src/main/java/com/farao_community/farao/util/CracFactorsProvider.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.util;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Cnec;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.NetworkElement;
+import com.farao_community.farao.data.crac_api.RangeAction;
+import com.powsybl.iidm.network.*;
+import com.powsybl.sensitivity.SensitivityFactor;
+import com.powsybl.sensitivity.SensitivityFactorsProvider;
+import com.powsybl.sensitivity.SensitivityFunction;
+import com.powsybl.sensitivity.SensitivityVariable;
+import com.powsybl.sensitivity.factors.BranchFlowPerInjectionIncrease;
+import com.powsybl.sensitivity.factors.BranchFlowPerPSTAngle;
+import com.powsybl.sensitivity.factors.functions.BranchFlow;
+import com.powsybl.sensitivity.factors.variables.InjectionIncrease;
+import com.powsybl.sensitivity.factors.variables.PhaseTapChangerAngle;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
+ */
+public class CracFactorsProvider implements SensitivityFactorsProvider {
+    private final Crac crac;
+
+    public CracFactorsProvider(Crac crac) {
+        this.crac = crac;
+    }
+
+    @Override
+    public List<SensitivityFactor> getFactors(Network network) {
```

Comment:
How the notion of State is handled by this method ? If I need to build sensitivity factors in order to compute the sensi on different network variants (N and N-1s) :

* the SensitivityFactors are not dependent from the State (they just use the information of network ID and names), and can therefore be used whatever the State for which we want to run the SensitivityAnalysis ?

* or, do I need to execute this getFactors() method on different Network, each network being updated with the topology of the N/N-1s State.

Reply:
Maybe we can add something in the javadoc on this.

Reply:
If you look at how the API of sensi is designed for systematic sensi analysis in powsybl-core snapshot, ContingenciesProvider and FactorsProvider are independant. We suppose that we want to observe same things in N and N-k, whichis not always correct.
I think the best is to have something that can work in any configuration (on any state).

And for the variant, the answer is still in the sensitivity computation API. In the run method, you provide the desired working variant. But as befoore, this will change a lot when upgrading powsybl version and using systematic computation....